lib/grammers-client/src/client/messages.rs

```diff
@@ -546,6 +546,119 @@ impl Client {
+        &self,
+        chat: C,
+        mut medias: Vec<InputMedia>,
+    ) -> Result<Vec<Message>, InvocationError> {
+        let chat = chat.into();
+        let random_ids = generate_random_ids(medias.len());
+
+        // Upload external files
+        for media in medias.iter_mut() {
+            let raw_media = media.media.clone().unwrap();
+
+            if matches!(
+                raw_media,
+                tl::enums::InputMedia::PhotoExternal(_)
+                    | tl::enums::InputMedia::DocumentExternal(_)
+            ) {
+                let uploaded = self
+                    .invoke(&tl::functions::messages::UploadMedia {
+                        business_connection_id: None,
+                        peer: chat.to_input_peer(),
+                        media: raw_media,
+                    })
+                    .await?;
+                media.media = Some(
+                    types::Media::from_raw(uploaded)
+                        .unwrap()
+                        .to_raw_input_media()
+                        .unwrap(),
+                );
+            }
+        }
+
+        let first_media = medias.first().unwrap();
+
+        let updates = self
+            .invoke(&tl::functions::messages::SendMultiMedia {
+                silent: false,
+                background: false,
+                clear_draft: false,
+                peer: chat.to_input_peer(),
+                reply_to: first_media.reply_to.map(|reply_to_msg_id| {
+                    tl::types::InputReplyToMessage {
+                        reply_to_msg_id,
+                        top_msg_id: None,
+                        reply_to_peer_id: None,
+                        quote_text: None,
+                        quote_entities: None,
+                        quote_offset: None,
+                    }
+                    .into()
+                }),
+                schedule_date: None,
+                multi_media: medias
+                    .iter()
+                    .zip(random_ids.iter())
+                    .map(|(input_media, random_id)| {
+                        let entities = parse_mention_entities(self, input_media.entities.clone());
+                        let raw_media = input_media.media.clone().unwrap();
+
+                        tl::enums::InputSingleMedia::Media(tl::types::InputSingleMedia {
+                            media: raw_media,
+                            random_id: *random_id,
+                            message: input_media.caption.clone(),
+                            entities,
+                        })
+                    })
+                    .collect(),
+                send_as: None,
+                noforwards: false,
+                update_stickersets_order: false,
+                invert_media: false,
+                quick_reply_shortcut: None,
+                effect: None,
+            })
+            .await?;
+
+        let mut messages = Vec::with_capacity(medias.len());
```

Comment:
…and if you use `into_iter`, simply take `random_ids.len()` here, which is the same.